oss-internship-2020/sapi_lodepng/examples/main_unit_test.cc

```diff
@@ -0,0 +1,189 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+#include "helpers.h"
+#include "sandbox.h"
+#include "sandboxed_api/util/status_matchers.h"
+
+using ::sapi::IsOk;
+using ::testing::Eq;
+using ::testing::IsTrue;
+using ::testing::NotNull;
+
+namespace {
+
+TEST(HelpersTest, CreateTempDirAtCWD) {
+  const std::string images_path = CreateTempDirAtCWD();
+  EXPECT_THAT(sandbox2::file_util::fileops::Exists(images_path, false),
+              IsTrue());
+
+  ASSERT_THAT(sandbox2::file_util::fileops::DeleteRecursively(images_path),
+              IsTrue());
+}
+
+TEST(HelpersTest, GenerateValues) {
+  EXPECT_THAT(GenerateValues().size(), Eq(kImgLen));
+}
+
+TEST(LodePngTest, Init) {
+  const std::string images_path = CreateTempDirAtCWD();
+  SapiLodepngSandbox sandbox(images_path);
+  ASSERT_THAT(sandbox.Init(), IsOk()) << "Error during sandbox init";
+
+  ASSERT_THAT(sandbox2::file_util::fileops::DeleteRecursively(images_path),
+              IsTrue());
+}
+
+// Generate an image, encode it, decode it and compare the pixels with the
+// initial values.
+TEST(LodePngTest, EncodeDecodeOneStep) {
+  const std::string images_path = CreateTempDirAtCWD();
+
+  SapiLodepngSandbox sandbox(images_path);
+  ASSERT_THAT(sandbox.Init(), IsOk()) << "Error during sandbox init";
+  LodepngApi api(&sandbox);
+
+  std::vector<uint8_t> image(GenerateValues());
+
+  sapi::v::Array<uint8_t> sapi_image(image.data(), kImgLen);
+  sapi::v::ConstCStr sapi_filename("/output/out_generated1.png");
+
+  SAPI_ASSERT_OK_AND_ASSIGN(
+      unsigned int result,
+      api.lodepng_encode32_file(sapi_filename.PtrBefore(),
+                                sapi_image.PtrBefore(), kWidth, kHeight));
+
+  ASSERT_THAT(result, Eq(0)) << "Result from encode32_file not 0";
```

Comment:
FYI: The error message generated by gtest will already show "expected" and "actual" value. So adding "is not 0" or sth is not necessary. "Unexpected result from encode32_file" would be cleaner in my opinion.